TST: add test for #9107

miggec opened the pull request:
test to ensure OutOfBoundsDatetime exception is raised when calling
pd.to_datetime with out-of-bounds date strings

As discussed briefly in #28367 

- [x] closes #9107 ? the issue is no longer truly reproducible, this PR adds a test around the new behaviour 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Commit:
add test for #9107

test to ensure OutOfBoundsDatetime exception is raised when calling
pd.to_datetime with out-of-bounds date strings

jbrockmendel commented on doc/source/whatsnew/v1.0.0.rst:
```diff
@@ -118,7 +118,7 @@ Datetimelike
 - Bug in :class:`Timestamp` subtraction when subtracting a :class:`Timestamp` from a ``np.datetime64`` object incorrectly raising ``TypeError`` (:issue:`28286`)
 - Addition and subtraction of integer or integer-dtype arrays with :class:`Timestamp` will now raise ``NullFrequencyError`` instead of ``ValueError`` (:issue:`28268`)
--
+- New test to ensure OutOfBoundsDatetime exception is raised when calling :func:``pd.to_datetime`` with out-of-bounds date strings
```
this note is not necessary since the change is not user-facing

TomAugspurger replied:
Though no real harm in including it. Your call @squigmig.

If you include it, you'll need to refer to an issue with :issue: and fix the formatting. Should be :func:\`to_datetime\` (single backticks, no `pd`).

jbrockmendel commented on pandas/tests/indexes/datetimes/test_tools.py:
```diff
@@ -1028,6 +1028,12 @@ def test_to_datetime_box_deprecated(self):
         assert result == expected
 
+    @pytest.mark.parametrize("dt_str", ["00010101", "13000101", "30000101", "99990101"])
+    def test_old_date_out_of_bounds_exception(self, dt_str):
```
a more informative name would be e.g. `test_to_datetime_with_format_out_of_bounds`

miggec committed:
Remove superfluous entry into whatsnew/v1.0.0.rst

New test is not user facing, does not need to be included in release notes

Commit:
Rename test to suggested, more informative name

test_old_date_out_of_bounds_exception -> test_to_datetime_with_format_out_of_bounds
